### PR TITLE
Fset-1379 Assessor enters scores and feedback

### PR DESCRIPTION
**Pull request by mgilgar**
3 PRs:
fset-faststream-admin-frontend
https://github.com/hmrc/fset-faststream-frontend/pull/215
https://github.com/hmrc/fset-faststream/pull/429

### CODE REVIEW

File: app/repositories/AssessmentScoresRepository.scala [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+  def findAll: Future[List[AssessmentScoresAllExercises]]
+}
+
+class AssessmentScoresMongoRepository(dateTime: DateTimeFactory)(implicit mongo: () => DB)
+  extends ReactiveRepository[AssessmentScoresAllExercises, BSONObjectID](CollectionNames.ASSESSMENT_SCORES, mongo,
+    AssessmentScoresAllExercises.jsonFormat, ReactiveMongoFormats.objectIdFormats)
+    with AssessmentScoresRepository with BaseBSONReader with ReactiveRepositoryHelpers {
+
+  // TODO: This save method does not remove exercise subdocument when in the case class they are None
```
Comment (kjharkin) [inline app/repositories/AssessmentScoresRepository.scala]: Please do something with this "TODO"

File: app/repositories/AssessmentScoresRepository.scala [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    val applicationId = allExercisesScores.applicationId.toString()
+    val query = BSONDocument("applicationId" -> applicationId)
+    val updateBSON = BSONDocument("$set" -> AssessmentScoresAllExercises.bsonHandler.write(allExercisesScores))
+    val validator = singleUpsertValidator(applicationId, actionDesc = "saving asessment scores")
+    collection.update(query, updateBSON, upsert = true) map validator
+  }
+
+  def find(applicationId: UniqueIdentifier): Future[Option[AssessmentScoresAllExercises]] = {
+    val query = BSONDocument("applicationId" -> applicationId.toString())
+    //collection.find(query, projection).one[FSACAllExercisesScoresAndFeedback]
```
Comment (kjharkin) [inline app/repositories/AssessmentScoresRepository.scala]: remove

File: app/repositories/AssessmentScoresRepository.scala [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    collection.find(query).one[BSONDocument].map(_.map(AssessmentScoresAllExercises.bsonHandler.read))
+  }
+
+  def findAll: Future[List[AssessmentScoresAllExercises]] = {
+    val query = BSONDocument.empty
+    //collection.find(query).cursor[FSACAllExercisesScoresAndFeedback](ReadPreference.nearest).collect[List]()
```
Comment (kjharkin) [inline app/repositories/AssessmentScoresRepository.scala]: remove

File: app/services/assessmentscores/AssessmentScoresService.scala [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+      applicationRepository.addProgressStatusAndUpdateAppStatus(
+        allExercisesScores.applicationId.toString(),
+        ProgressStatuses.ASSESSMENT_CENTRE_SCORES_ENTERED)
+    } else {
+      Future.successful(())
+    }
+  }
+
+  def findAssessmentScoresWithCandidateSummaryByApplicationId(applicationId: UniqueIdentifier): Future[AssessmentScoresFindResponse] = {
+    // TODO: return Option. if assessmentScores are not found, return None
```
Comment (kjharkin) [inline app/services/assessmentscores/AssessmentScoresService.scala]: Please do something with this TODO

**Comment by kjharkin** [comment]
Commented out code seems to be common with this ticket. Kindly check that these have been removed before PRs get reviewed

File: app/repositories/AssessmentScoresRepository.scala (hunk 1 above)
Comment (mgilgar) [inline app/repositories/AssessmentScoresRepository.scala]: Given that I will continue working in this area later on, it is a good reminder to me.

File: app/repositories/AssessmentScoresRepository.scala (hunk 3 above)
Comment (mgilgar) [inline app/repositories/AssessmentScoresRepository.scala]: I will remove.

File: app/services/assessmentscores/AssessmentScoresService.scala (hunk 4 above)
Comment (mgilgar) [inline app/services/assessmentscores/AssessmentScoresService.scala]: We will need more time to handle that and I think we want this in as soon as possible.

**Pull request merged**